include/aws/io/retry_strategy.h

```diff
@@ -108,8 +113,20 @@ struct aws_exponential_backoff_retry_options {
     /** Jitter mode to use, see comments for aws_exponential_backoff_jitter_mode.
      * Default is AWS_EXPONENTIAL_BACKOFF_JITTER_DEFAULT */
     enum aws_exponential_backoff_jitter_mode jitter_mode;
-    /** By default this will be set to use aws_device_random. If you want something else, set it here. */
+
+    /** Deprecated. Use generate_random_impl instead
+     * By default this will be set to use aws_device_random. If you want something else, set it here.
+     * */
     uint64_t (*generate_random)(void);
+
+    /*
+     * By default this will be set to use aws_device_random. If you want something else, set it here.
+     */
+    generate_random_fn *generate_random_impl;
```

Comment:
ah, the eternal "what do I call v2" question

Reply:
Any better ideas?